apps/builder/app/builder/features/style-panel/sections/box-shadows/box-shadow-layer.tsx

```diff
@@ -0,0 +1,117 @@
+import type { LayersValue, TupleValue } from "@webstudio-is/css-data";
+import {
+  Label,
+  SmallIconButton,
+  SmallToggleButton,
+  CssValueListItem,
+  styled,
+  theme,
+} from "@webstudio-is/design-system";
+import {
+  EyeconClosedIcon,
+  EyeconOpenIcon,
+  SubtractIcon,
+} from "@webstudio-is/icons";
+import { useMemo } from "react";
+import { FloatingPanel } from "~/builder/shared/floating-panel";
+import { BoxShadowContent } from "./box-shadow-content";
+import type { RenderCategoryProps } from "../../style-sections";
+import { colord } from "colord";
+import { toValue } from "@webstudio-is/css-engine";
+
+const LayerThumbnail = styled("div", {
+  width: theme.spacing[10],
+  height: theme.spacing[10],
+});
+
+const useLayer = (layer: TupleValue) => {
+  return useMemo(() => {
+    const name = [];
+    const shadow = [];
+    let color =
+      "repeating-conic-gradient(rgba(0, 0, 0, 0.22) 0%, rgba(0, 0, 0, 0.22) 25%, transparent 0%, transparent 50%) 0% 33.33% / 40% 40%";
+
+    for (const item of Object.values(layer.value)) {
+      if (item.type === "unit") {
+        const value = toValue(item);
+        name.push(value);
+        shadow.push(value);
+      }
+
+      if (item.type === "rgb") {
+        color = toValue(item);
+        shadow.push(toValue(item));
+      }
+
+      if (item.type === "keyword") {
+        if (colord(item.value).isValid() === false) {
+          name.push(item.value);
+        } else {
+          color = item.value;
+        }
+        shadow.push(item.value);
+      }
+    }
+
+    return [name.join(" "), shadow.join(" "), color];
+  }, [layer]);
+};
+
+export const Layer = (
+  props: Pick<RenderCategoryProps, "createBatchUpdate"> & {
+    id: string;
+    index: number;
+    layer: TupleValue;
+    isHighlighted: boolean;
+    onLayerHide: (index: number) => void;
+    onDeleteLayer: (index: number) => void;
+    onEditLayer: (index: number, layers: LayersValue) => void;
+  }
+) => {
+  const { index, id, layer, isHighlighted, onDeleteLayer, onLayerHide } = props;
+  const [layerName, shadow, color] = useLayer(layer);
+
+  return (
+    <FloatingPanel
+      title="Box Shadow"
+      content={
+        <BoxShadowContent
+          index={index}
+          value={shadow}
+          onEditLayer={props.onEditLayer}
+          createBatchUpdate={props.createBatchUpdate}
+        />
+      }
+    >
+      <CssValueListItem
+        id={id}
+        active={isHighlighted}
+        index={index}
+        label={<Label truncate>{layerName}</Label>}
+        hidden={layer?.hidden}
+        thumbnail={<LayerThumbnail css={{ background: color }} />}
```

Comment:
1. we have the same thing in color picker and should reuse them, e.g. ColorThumbnail
2. ColorThumbnail can have a color prop and have a default that is chessboard
3. this is already wrong, because its missing the rounded corners

Reply:
Just used the same from the `background-gradients`. Will. change this, btw i don't see any component like `ColorThumbnail` in `ColorPicker`. Does it need to be extracted or something ?

Reply:
yes